### ISSUE TITLE
Support custom cloudtrail namespaces

### ISSUE THREAD

**Issue by danielpops** [duo-labs/cloudtracker#6]
We already ingest cloudtrail data to an elasticsearch cluster, but we have the data indexed in a nested dictionary structure for namespacing reasons.

For example, if a cloudtrail json blob looks roughly like: 

```
{
  "eventVersion": "1.0",
  "userIdentity": {
    "type": "IAMUser",
    "principalId": "EX_PRINCIPAL_ID",
    "arn": "arn:aws:iam::123456789012:user/Alice",
    "accessKeyId": "EXAMPLE_KEY_ID",
    "accountId": "123456789012",
    "userName": "Alice"
  },
  "eventTime": "2014-03-06T21:22:54Z",
  "eventSource": "ec2.amazonaws.com",
  "eventName": "StartInstances",
  "awsRegion": "us-east-2",
  "sourceIPAddress": "205.251.233.176",
  "userAgent": "ec2-api-tools 1.6.12.2",
  "requestParameters": {
    "instancesSet": {
      "items": [
        {
          "instanceId": "i-ebeaf9e2"
        }
      ]
    }
  },
  "responseElements": {
    "instancesSet": {
      "items": [
        {
          "instanceId": "i-ebeaf9e2",
          "currentState": {
            "code": 0,
            "name": "pending"
          },
          "previousState": {
            "code": 80,
            "name": "stopped"
          }
        }
      ]
    }
  }
}
```

It would actually look like:
```
{
  "the_cloudtrail_data": {
    "eventVersion": "1.0",
    "userIdentity": {
      "type": "IAMUser",
      "principalId": "EX_PRINCIPAL_ID",
      "arn": "arn:aws:iam::123456789012:user/Alice",
      "accessKeyId": "EXAMPLE_KEY_ID",
      "accountId": "123456789012",
      "userName": "Alice"
    },
    "eventTime": "2014-03-06T21:22:54Z",
    "eventSource": "ec2.amazonaws.com",
    "eventName": "StartInstances",
    "awsRegion": "us-east-2",
    "sourceIPAddress": "205.251.233.176",
    "userAgent": "ec2-api-tools 1.6.12.2",
    "requestParameters": {
      "instancesSet": {
        "items": [
          {
            "instanceId": "i-ebeaf9e2"
          }
        ]
      }
    },
    "responseElements": {
      "instancesSet": {
        "items": [
          {
            "instanceId": "i-ebeaf9e2",
            "currentState": {
              "code": 0,
              "name": "pending"
            },
            "previousState": {
              "code": 80,
              "name": "stopped"
            }
          }
        ]
      }
    }
  }
}
```

I think cloudtracker could expose the concept of a `key_prefix` or `key_namespace` or something, to account for this type of setup

**Comment by 0xdabbad00** [comment]
Makes sense.  I need to make the index configurable, as right now it is hard-coded at: https://github.com/duo-labs/cloudtracker/blob/07dcc7697e39876e6acdaf2b35a69f61448b0cdc/cloudtracker/datasources/es.py#L22

And then I think I could support the `key_prefix` by just prefixing the various query fields, for example this: https://github.com/duo-labs/cloudtracker/blob/07dcc7697e39876e6acdaf2b35a69f61448b0cdc/cloudtracker/datasources/es.py#L52

**Comment by danielpops** [comment]
Ack; that's exactly what I had in mind. I'm eager to run this against my cloudtrail data set, so I'll hopefully be adding this shortly :)

**Comment by danielpops** [comment]
Actually there is at least one issue. You are not considering key_prefix for the `filter_errors` field name. You should do something like 

```
$ git diff -U10
diff --git a/cloudtracker/datasources/es.py b/cloudtracker/datasources/es.py
index 9b58e48..9cc8795 100644
--- a/cloudtracker/datasources/es.py
+++ b/cloudtracker/datasources/es.py
@@ -29,21 +29,21 @@ class ElasticSearch(object):
     def __init__(self, config, start, end):
         # Open connection to ElasticSearch
         self.es = Elasticsearch([config], timeout=900)
         self.searchfilter = {}
         self.index = config.get('index', 'cloudtrail')
         self.key_prefix = config.get('key_prefix', '')
         if self.key_prefix != "":
             self.key_prefix += "."

         # Filter errors
-        self.searchfilter['filter_errors'] = ~Q('exists', field='errorCode')
+        self.searchfilter['filter_errors'] = ~Q('exists', field='{0}errorCode'.format(self.key_prefix))

         # Filter dates
         if start:
             self.searchfilter['start_date_filter'] = Q('range', eventTime={'gte': start})
         if end:
             self.searchfilter['end_date_filter'] = Q('range', eventTime={'lte': end})

     def get_field_name(self, field):
         return self.key_prefix + field + ".keyword"
```

**Comment by danielpops** [comment]
hmmmm there's another problem. The `start_date` and `end_date` range queries are not using the `key_prefix`, which is more annoying to address because `eventTime` is being passed as a kwarg (although, on second thought, this may actually be a different issue in that the user should probably be able to specify which which field is being used as the index's timestamp (in my case, it's not `eventTime`, and in fact, it's not even `the_cloudtrail_data.eventTime`))

**Comment by 0xdabbad00** [comment]
@danielpops Is this resolved now with the work you did for #10?

**Comment by danielpops** [comment]
Yup. This appears to be working as expected. Thanks alot!
